### PR TITLE
Add Mutation Count and Fraction Genome Altered as clinical data

### PR DESCRIPTION
**Pull request by ersinciftci**
And delete `mutation_count` and `fraction_genome_altered` tables, and delete API endpoints using those table.

Fixes https://github.com/cBioPortal/cbioportal/issues/4631

### CODE REVIEW

File: core/src/main/java/org/mskcc/cbio/portal/scripts/ImportCopyNumberSegmentData.java [hunk 1]
```diff
@@ -144,7 +144,14 @@ public void run() {
     
     private void importFractionGenomeAltered(CancerStudy cancerStudy) throws DaoException {
 
-        DaoCopyNumberSegment.calculateFractionGenomeAltered(cancerStudy.getInternalId());
+        ClinicalAttribute attr = new ClinicalAttribute("FRACTION_GENOME_ALTERED", "Fraction Genome Altered", "Fraction Genome Altered", "NUMBER",
+            false, "1", cancerStudy.getInternalId());
```
Comment (jjgao) [inline core/src/main/java/org/mskcc/cbio/portal/scripts/ImportCopyNumberSegmentData.java]: @zhx828 what's the priority for "FRACTION_GENOME_ALTERED" in the study view?

File: core/src/main/java/org/mskcc/cbio/portal/scripts/ImportExtendedMutationData.java [hunk 2]
```diff
@@ -431,41 +431,48 @@ public void importData() throws IOException, DaoException {
...
+        for (ExtendedMutation mutation : mutations.values()) {
+            try {
+                DaoMutation.addMutation(mutation,false);
+            } catch (DaoException ex) {
+                throw ex;
+            }
+        }
+
+        ClinicalAttribute attr = new ClinicalAttribute("MUTATION_COUNT", "Mutation Count", "Mutation Count", "NUMBER",
+            false, "1", geneticProfile.getCancerStudyId());
```
Comment (jjgao) [inline core/src/main/java/org/mskcc/cbio/portal/scripts/ImportExtendedMutationData.java]: @zhx828 what's the priority for "MUTATION_COUNT" in the study view?

File: core/src/main/java/org/mskcc/cbio/portal/scripts/ImportExtendedMutationData.java [hunk 3]
```diff
@@ -431,41 +431,48 @@ public void importData() throws IOException, DaoException {
...
+        DaoClinicalAttributeMeta.addDatum(attr);
+        
+        Map<Integer, String> mutationCounts = DaoMutation.getMutationCounts(geneticProfileId);
+        for (Map.Entry<Integer, String> mutationCount : mutationCounts.entrySet()) {
+            DaoClinicalData.addSampleDatum(mutationCount.getKey(), "MUTATION_COUNT", mutationCount.getValue());
```
Comment (jjgao) [inline core/src/main/java/org/mskcc/cbio/portal/scripts/ImportExtendedMutationData.java]: Maybe create a function `DaoMutation.createMutationCountClinicalData(geneticProfileId)` (similar to `DaoMutation.calculateMutationCountByKeyword(geneticProfileId)`) and moved lines 450-457 there? It maybe also good to make `DaoMutation.getMutationCounts` private or even move the code inside `createMutationCountClinicalData`.

File: web/src/main/java/org/cbioportal/web/MutationSpectrumController.java [hunk 4]
```diff
@@ -43,15 +43,15 @@
         @ApiParam(required = true, value = "List of Sample IDs/Sample List ID")
         @Valid @RequestBody MutationSpectrumFilter mutationSpectrumFilter) throws MolecularProfileNotFoundException {
 
-        List<MutationSpectrum> fractionGenomeAlteredList;
+        List<MutationSpectrum> mutationSpectrums;
```
Comment (jjgao) [inline web/src/main/java/org/cbioportal/web/MutationSpectrumController.java]: good catch.

File: db-scripts/src/main/resources/migration.sql [hunk 5]
```diff
@@ -457,3 +457,23 @@ ALTER TABLE gistic_to_gene ADD CONSTRAINT `gistic_to_gene_ibfk_2` FOREIGN KEY (`
 ALTER TABLE `mutation_event` MODIFY COLUMN `KEYWORD` VARCHAR(255);
 ALTER TABLE `mutation_count_by_keyword` MODIFY COLUMN `KEYWORD` VARCHAR(255);
 UPDATE `info` SET `DB_SCHEMA_VERSION`="2.6.1";
+
+##version: 2.7.0
+INSERT INTO `clinical_attribute_meta` SELECT 'MUTATION_COUNT', 'Mutation Count', 'Mutation Count', 'NUMBER', 0, '1',
```
Comment (jjgao) [inline db-scripts/src/main/resources/migration.sql]: Please find out the priority value that we should use.

File: db-scripts/src/main/resources/migration.sql [hunk 6]
```diff
@@ -457,3 +457,23 @@ ALTER TABLE gistic_to_gene ADD CONSTRAINT `gistic_to_gene_ibfk_2` FOREIGN KEY (`
...
+##version: 2.7.0
+INSERT INTO `clinical_attribute_meta` SELECT 'MUTATION_COUNT', 'Mutation Count', 'Mutation Count', 'NUMBER', 0, '1', 
+genetic_profile.`CANCER_STUDY_ID` FROM mutation_count INNER JOIN genetic_profile ON 
+mutation_count.`GENETIC_PROFILE_ID` = genetic_profile.`GENETIC_PROFILE_ID` GROUP BY genetic_profile.`CANCER_STUDY_ID`;
+
+INSERT INTO `clinical_sample` SELECT mutation_count.`SAMPLE_ID`, 'MUTATION_COUNT', mutation_count.`MUTATION_COUNT` 
+FROM mutation_count;
+
+INSERT INTO `clinical_attribute_meta` SELECT 'FRACTION_GENOME_ALTERED', 'Fraction Genome Altered', 
+'Fraction Genome Altered', 'NUMBER', 0, '1', fraction_genome_altered.`CANCER_STUDY_ID` FROM fraction_genome_altered
```
Comment (jjgao) [inline db-scripts/src/main/resources/migration.sql]: Please find out the priority value that we should use.

File: core/src/main/java/org/mskcc/cbio/portal/dao/DaoMutation.java [hunk 7]
```diff
@@ -130,21 +130,25 @@ public static int addMutationEvent(ExtendedMutation.MutationEvent event) throws
         return 1;
     }
 
-    public static int calculateMutationCount (int profileId) throws DaoException {
+    public static Map<Integer, String> getMutationCounts(int profileId) throws DaoException {
```
Comment (jjgao) [inline core/src/main/java/org/mskcc/cbio/portal/dao/DaoMutation.java]: make this private or remove - see comments below.

File: core/src/main/java/org/mskcc/cbio/portal/dao/DaoCopyNumberSegment.java [hunk 8]
```diff
@@ -66,23 +66,27 @@ public static int addCopyNumberSegment(CopyNumberSegment seg) throws DaoExceptio
         }
     }
 
-    public static int calculateFractionGenomeAltered(int cancerStudyId) throws DaoException {
+    public static Map<Integer, String> getFractionGenomeAltered(int cancerStudyId) throws DaoException {
```
Comment (jjgao) [inline core/src/main/java/org/mskcc/cbio/portal/dao/DaoCopyNumberSegment.java]: make this function private. See comments below.

File: core/src/main/java/org/mskcc/cbio/portal/scripts/ImportCopyNumberSegmentData.java [hunk 9]
```diff
@@ -144,7 +147,17 @@ public void run() {
...
 
-        DaoCopyNumberSegment.calculateFractionGenomeAltered(cancerStudy.getInternalId());
+        ClinicalAttribute clinicalAttribute = DaoClinicalAttributeMeta.getDatum(FRACTION_GENOME_ALTERED_ATTR_ID, cancerStudy.getInternalId());
+        if (clinicalAttribute == null) {
+            ClinicalAttribute attr = new ClinicalAttribute(FRACTION_GENOME_ALTERED_ATTR_ID, "Fraction Genome Altered", "Fraction Genome Altered", "NUMBER",
+                false, "1", cancerStudy.getInternalId());
+            DaoClinicalAttributeMeta.addDatum(attr);
+        }
+        
+        Map<Integer, String> mutationCounts = DaoCopyNumberSegment.getFractionGenomeAltered(cancerStudy.getInternalId());
```
Comment (jjgao) [inline core/src/main/java/org/mskcc/cbio/portal/scripts/ImportCopyNumberSegmentData.java]: Maybe create a function `DaoMutation.createFractionGenomeAlteredClinicalData(cancerStudyId)` and moved lines 150-160 there? It maybe also good to make `DaoCopyNumberSegment.getFractionGenomeAltered` private or even move the code inside `createFractionGenomeAlteredClinicalData`.

File: core/src/main/java/org/mskcc/cbio/portal/scripts/ImportExtendedMutationData.java [hunk 10]
```diff
@@ -431,41 +433,51 @@ public void importData() throws IOException, DaoException {
...
+        }
+
+        ClinicalAttribute clinicalAttribute = DaoClinicalAttributeMeta.getDatum(MUTATION_COUNT_ATTR_ID, geneticProfile.getCancerStudyId());
+        if (clinicalAttribute == null) {
+            ClinicalAttribute attr = new ClinicalAttribute(MUTATION_COUNT_ATTR_ID, "Mutation Count", "Mutation Count", "NUMBER",
+                false, "1", geneticProfile.getCancerStudyId());
+            DaoClinicalAttributeMeta.addDatum(attr);
+        }
+        
+        Map<Integer, String> mutationCounts = DaoMutation.getMutationCounts(geneticProfileId);
```
Comment (jjgao) [inline core/src/main/java/org/mskcc/cbio/portal/scripts/ImportExtendedMutationData.java]: make `DaoMutation.getMutationCounts` private and create another function `DaoMutation.createMutationCountClinicalData(geneticProfileId)`

File: db-scripts/src/main/resources/migration.sql (hunk 5 above)
Comment (zhx828) [reply db-scripts/src/main/resources/migration.sql]: @jjgao @ersinciftci according to https://github.com/cBioPortal/cbioportal/blob/master/docs/Study-View.md, let's set Mutation Count as 30, FGA as 20

File: core/src/main/java/org/mskcc/cbio/portal/scripts/ImportCopyNumberSegmentData.java (hunk 1 above)
Comment (ersinciftci) [reply core/src/main/java/org/mskcc/cbio/portal/scripts/ImportCopyNumberSegmentData.java]: https://github.com/cBioPortal/cbioportal/blob/master/portal/src/main/webapp/js/src/dashboard/resources/configs.json#L29 seems like it's 20

File: core/src/main/java/org/mskcc/cbio/portal/scripts/ImportExtendedMutationData.java (hunk 2 above)
Comment (ersinciftci) [reply core/src/main/java/org/mskcc/cbio/portal/scripts/ImportExtendedMutationData.java]: https://github.com/cBioPortal/cbioportal/blob/master/portal/src/main/webapp/js/src/dashboard/resources/configs.json#L28 seems like it's 30

**Pull request merged**